### PR TITLE
Add stackoverflow profile option

### PR DESCRIPTION
**Pull request by scott-mcnulty**
Adding an option to link a stackoverflow profile story

### CODE REVIEW

**Comment by sproogen** [comment]
Thanks for the PR.
I'm not sure on just linking to the stack overflow story, I feel like it should be a bit clearer in the config that is a link to the story.
Possibly it could optionally link to the main stack overflow profile page or the story.

**Comment by EndermanAPM** [comment]
As far as I know the user id is needed to navigate to the profile but the story lets you set a public (static) route. (And that's the value that you set in the config)

File: _includes/header.html [hunk 1]
```diff
@@ -83,6 +83,13 @@ <h2>{{ site.title | escape }}</h2>
             </a>
           </li>
         {%- endif -%}
+        {%- if site.stackoverflow_username -%}
+          <li>
+            <a target="_blank" href="https://stackoverflow.com/story/{{ site.stackoverflow_username| cgi_escape | escape }}" class="button button--sacnite button--round-l">
+              <i class="fa fa-stack-overflow" title="Stackoverflow link"></i>
```
Comment (mrufflesmcghie) [inline _includes/header.html]: Should we change this to `"fab fa-stack-overflow"` due to upgrade of Font Awesome #65

**Comment by sproogen** [comment]
Added a stack overflow link in #76

**Pull request closed without merging**